extensions/wikia/UserProfilePageV3/maintenance/migrateAvatarsViaService.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * Script migrates the user avatars from DFS to user avatars service
+ *
+ * @see PLATFORM-1419
+ *
+ * @author Macbre
+ * @ingroup Maintenance
+ */
+
+putenv( 'SERVER_ID=177' ); // run in the context of a community wiki
+
+require_once( __DIR__ . '/../../../../maintenance/Maintenance.php' );
+
+class AvatarsMigratorException extends Exception {}
+
+/**
+ * Maintenance script class
+ */
+class AvatarsMigrator extends Maintenance {
+
+	private $isDryRun = true;
+
+	/**
+	 * Set script options
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->addOption('dry-run', 'Don\'t perform any operations [default]');
+		$this->addOption( 'force', 'Apply the changes made by the script' );
+
+		$this->mDescription = 'This script migrates the user avatars from DFS to user avatars service';
+	}
+
+	public function execute() {
+		// read options
+		$this->isDryRun = $this->hasOption( 'dry-run' ) || !$this->hasOption( 'force' );
+
+		if ($this->isDryRun) $this->output( "Running in dry-run mode!\n\n" );
+
+		$this->output( "Getting the list of all accounts...\n" );
+
+		// get all accounts
+		global $wgExternalSharedDB;
+		$db = $this->getDB( DB_SLAVE, [], $wgExternalSharedDB );
+
+		$res = $db->select(
+			'`user`',
+			'user_id AS id',
+			[],
+			__METHOD__
+		);
```

Comment:
@macbre as far as I know, yes, the preferences service will keep storing preferences in the `user_preferences`. I haven't heard any plans to migrate that data to another table yet. Especially not in the short term. So yes, performing that query will get a list of all accounts that requires an avatar migration.